client/authority-discovery/src/lib.rs

```diff
@@ -377,10 +377,12 @@ where
 						"Failed to put hash '{:?}' on Dht.", hash
 					)
 				},
+				// The sender side of the dht event stream has been closed, likely due to the
```

Comment:
```suggestion
				// The sender side of the dht event stream has been closed, _likely_ due to the
```

why can't isn't it the case that the underlying component (network or ...) can't explicitly say "I am dying" rather than relying on _likeliness_ of `Ready(None)` to mean it is terminated? 

(I am not familiar with these code at all, I might be very out of scope :D)

Reply:
I think the notion of dropping a sender to signal the receiver its termination is a common mechanism which we make use of a lot within `client/network`.

Why did I write `likely`? There could be other reasons, e.g. where parts of the network paniced. Thus I wanted to make sure someone debugging such failure does not take sender dropping as network termination for granted.

@kianenigma does the above make sense? Was the above suggestion meant to be included or just a pointer to your question?